### PR TITLE
rel: prep beta release 0.1.2b0

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- lack of a published beta package

## Short description of the changes

- update version in `pyproject.toml` to 0.1.0b0
- add notes in `CHANGELOG.md` for beta release
- small cleanup throughout (typos, formatting, etc to clear up squigglies from my editor)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # honeycomb-opentelemetry-python changelog
 
+## [0.1.0b0] - 2023-03-29
+
+Initial beta release of Honeycomb's OpenTelemetry distribution for Python!
+
+### Maintenance
+
+- ci: require smoke tests for publish steps (#107) | [@pkanal](https://github.com/pkanal)
+- maint: drop poetry locks from example apps (#111) | [@JamieDanielson](https://github.com/JamieDanielson)
```
Comment (robbkidd) [inline CHANGELOG.md:10]: Ride the Lightning!

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "honeycomb-opentelemetry"
-version = "0.1.1a3"
+version = "0.1.0b0"
```
Comment (robbkidd) [inline pyproject.toml:3]: 🤔   I was expecting maybe `0.1.2b0` or `0.2.0b0`. Why go downwards with the patch version?

**Review by robbkidd — COMMENTED** [review]
A question about the next version number …

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,16 @@
 # honeycomb-opentelemetry-python changelog
 
+## [0.1.0b0] - 2023-03-29
```
Comment (robbkidd) [inline CHANGELOG.md:3]:
```suggestion
## [0.1.2b0] - 2023-03-29
```

File: pyproject.toml (hunk 2 above)
Comment (robbkidd) [inline pyproject.toml:3]:
```suggestion
version = "0.1.2b0"
```

**Review by robbkidd — CHANGES_REQUESTED** [review]
Here's a proposed patch bump from 0.1.1a3 to 0.1.2b0 coming in as suggested edits that can be accepted in one commit. If accepted, we can update the PR title and ignore the branch name.

**Review by MikeGoldsmith — APPROVED** [review]
🎉

**Pull request merged**